21 api crud room comment

buihieu028 opened the pull request:
Bổ sung CRUD cho room comment

GuchiGangz committed:
add/RoomComment module

Bổ sung repository và interface cho RoomComment module

GuchiGangz committed:
add/request cho từng method

Bổ sung FormRequest cho RoomCommentRequest(AddNewComment)

DeleteCommentRequest(DeleteComment)

GuchiGangz committed:
add/JsonResource

Bổ sung resource trả về ở AddComment và UpdateComment

GuchiGangz committed:
add/RoomComment controller

Tạo controller cho room-comment

DinnieJ commented on api/routes/api.php:
```diff
@@ -31,6 +31,9 @@
 });
 Route::group(['prefix' => 'tenant'], function () {
     Route::get('test', 'TestController@tenantTest')->middleware(['auth.jwt', 'assign.guard:tenant']);
+    Route::post('/addcomment', 'RoomComment\RoomCommentController@addNewComment');
```
sửa lại routing đi, để thành /api/tenant/comment/add chứ k /api/tenant/addcomment

DinnieJ commented on api/app/Http/Controllers/RoomComment/RoomCommentController.php:
```diff
@@ -0,0 +1,76 @@
+        return response()->json(null, 404);
+    }
+
+    public function getComment()
```
Hàm trắng thì vứt đi, đừng để rác

GuchiGangz committed:
update/api comment

Sửa lại URL cho các function của CRUD comment

Loại bỏ hàm thừa getComment() vì trong roomDetail đã display comment ra rồi
